### PR TITLE
ensure that mongodb is running

### PR DESCRIPTION
**Pull request by mostafahussein**
This PR adds a check for mongodb to ensure that the container is running before continue the provision process

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @mostafahussein! I've created [OSPR-3870](https://openedx.atlassian.net/browse/OSPR-3870) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

**Comment by natabene** [comment]
@mostafahussein Thank you for your contribution. @jmbowman This is ready for your review.

File: provision.sh [hunk 1]
```diff
@@ -28,11 +28,21 @@ do
   sleep 1
 done
 
+# Ensure the MongoDB server is online and usable
+echo "Waiting for MongoDB"
+until docker exec -i edx.devstack.mongo mongo --eval "printjson(db.serverStatus())" &> /dev/null
+do
+  printf "."
+  sleep 1
+done
```
Comment (jmbowman) [inline provision.sh:43]: Could you move this block below `echo -e "MySQL ready"` below?  MongoDB is usually up and running  by the end of the 20-second sleep below, whereas putting this here might slow down startup a little.

**Comment by natabene** [comment]
@mostafahussein Are you still interested in pursuing this and implementing Jeremy's feedback?

File: provision.sh (hunk 1 above)
Comment (mostafahussein) [reply provision.sh:43]: Updated

**Comment by mostafahussein** [comment]
@natabene Sorry for the late follow-up, I have updated the code as @jmbowman suggested.

**Review by jmbowman — APPROVED** [review]
Thanks!

**Comment by mostafahussein** [comment]
@jmbowman the current ci error is not related to the pr modification itself

**Comment by jmbowman** [comment]
The Travis failure is a pre-existing problem with the job taking a little too long, merging.

**Comment by openedx-webhooks** [comment]
@mostafahussein 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/devstack&entry.867055334=https://github.com/edx/devstack/pull/443&entry.1484655318=https://github.com/mostafahussein&entry.752974735=2019-10-03+00:26&entry.1917517419=2019-10-28+15:00&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**